src/lib/aws.ts

```diff
@@ -140,45 +141,77 @@ export async function runECSFargateTask(
     return result
 }
 
+async function getResourceCommandWrapper(
+    tagFilters: TagFilter[],
+    resourceTypeFilters: string[],
+    client: ResourceGroupsTaggingAPIClient,
+    logMessage: string,
+): Promise<ResourceTagMapping[]> {
+    let paginationToken = ''
+    let accumulatedResources: ResourceTagMapping[] | undefined = undefined
+    while (true) {
+        const command = new GetResourcesCommand({
+            TagFilters: tagFilters,
+            ResourceTypeFilters: resourceTypeFilters,
+            PaginationToken: paginationToken,
+            ResourcesPerPage: 2,
+        })
+
+        console.log(`AWS: START: ${logMessage}`)
+        const result = await client.send(command)
+        // Throw if values are unexpectedly undefined
+        result.ResourceTagMappingList = ensureValueWithError(result.ResourceTagMappingList)
+        result.PaginationToken = ensureValueWithError(result.PaginationToken)
+        console.log(`AWS:   END: GetResourcesCommand finished w/ list`, result.ResourceTagMappingList)
+        // move onto the next page if there is one
+        paginationToken = result.PaginationToken
+        accumulatedResources = [...(accumulatedResources || []), ...result.ResourceTagMappingList]
+        if (paginationToken === '') {
+            break
+        } else {
+            console.log('AWS:        There are more pages of results, getting next page ...')
+        }
+    }
+    console.log('AWS:        Found all resources matching input')
+    return accumulatedResources
+}
+
 export async function getTaskResourcesByRunId(
```

Comment:
It looks like this function isn't used anymore, so I just wanted to confirm we're just leaving it here in case it becomes useful in the future?

Reply:
basically yeah. i can see an argument for either Retain or Remove. what do you think?

Reply:
It's not a lot of code + it relies on a shared helper, so it doesn't add much in terms of maintenance. Maybe keep it around for now and in the future we can remove it if we feel like it's definitely never going to get used :man_shrugging: ?